src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -124,19 +125,21 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 		log.Error(err, "Failed to unmark node as worker")
 	}
 
-	log.Info("Stopping worker services")
-	if err := snaputil.StopWorkerServices(ctx, snap); err != nil {
-		log.Error(err, "Failed to stop worker services")
-	}
-
 	log.Info("Cleaning up control plane certificates")
 	if _, err := setup.EnsureControlPlanePKI(snap, &pki.ControlPlanePKI{}); err != nil {
 		log.Error(err, "failed to cleanup control plane certificates")
 	}
 
-	log.Info("Stopping control plane services")
-	if err := snaputil.StopControlPlaneServices(ctx, snap); err != nil {
-		log.Error(err, "Failed to stop control-plane services")
+	if _, ok := cfg.Annotations[apiv1.AnnotationSkipStopServicesOnRemove]; !ok {
+		log.Info("Stopping worker services")
+		if err := snaputil.StopWorkerServices(ctx, snap); err != nil {
```

Comment:
can we have some sort of a condition to see if this is a worker or a controlplane node and then stop the related services? 
not that it's super important, but I think it clarifies things a bit more.

Reply:
We explicitly don't want this. 
We check if a node is a worker by checking if the `worker` marker file exists. This can potentially fail for various IO reasons.
This is the remove hook, so everything we do is on a `best-effort` basis. If stopping the worker services fails (e.g. because the node is a CP) then this is fine and we just continue.